lib/pf/enforcement.pm

```diff
@@ -173,7 +173,13 @@ sub _vlan_reevaluation {
 
     if (isenabled($switch->{_deauthOnPrevious})) {
         $locationlog_entry = locationlog_last_entry_previous_switch($mac,$switch);
-        $switch = pf::SwitchFactory->instantiate( { switch_mac => $locationlog_entry->{'switch_mac'}, switch_ip => $locationlog_entry->{'switch_ip'} } );
+        if ($locationlog_entry) {
```

Comment:
We should have a warning that no location entry was found and that the deauth will not happen

Reply:
Do we also skip the deauth on the current switch?

Reply:
I think @fdurand will answer better than me but I'd say we don't deauth on the current switch if it's configured to be the previous one and there isn't a way to do it. Otherwise it doesn't respect the behavior that is configured.

Reply:
Is a warning needed? Since that scenario can happen in a normal workflow I can see how it would be useful to know if it happened.

Reply:
Yes the goal of that is to be able to deauth on the previous equipment. Like you have 2 equipments and the only way to reevaluate the acces is to do the deauth on the n-1 then do it if possible , otherwise do nothing and log a warning.

Reply:
Ok done please review